MapleServer2/AIScripts/PrisonGovernor2Battle.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Maple2Storage.Types;
+using MapleServer2.Data.Static;
+using MapleServer2.Managers.Actors;
+using MapleServer2.Types;
+
+namespace MapleServer2.AIScripts;
+
+public class PrisonGovernor2Battle : AIState
+{
+    public override void OnEnter(Npc npc)
+    {
+        // Ok to assume this will only happen on tutorial, so always attack the only player available
+        npc.Target = npc.FieldManager!.State.Players.FirstOrDefault().Value;
+    }
+
+    public override AIState? Execute(Npc npc)
+    {
+        if (npc.CheckHPThreshold(50))
+        {
+            return new BattleStop();
+        }
+
+        float playerMobDist = CoordF.Distance(npc.Target!.Coord, npc.Coord);
+
+        if (playerMobDist > npc.Value.NpcMetadataDistance.Sight)
+        {
+            return null;
+        }
+
+        if (playerMobDist >= npc.Value.NpcMetadataDistance.Avoid * 2)
+        {
+            npc.Animation = AnimationStorage.GetSequenceIdBySequenceName(npc.Value.NpcMetadataModel.Model, "Run_A");
+
+            // Get coord between the player and the mob that is the avoid distance
+            CoordF coord = CoordF.From(
+                npc.Target.Coord.X + (npc.Coord.X - npc.Target.Coord.X) * npc.Value.NpcMetadataDistance.Avoid / playerMobDist,
+                npc.Target.Coord.Y + (npc.Coord.Y - npc.Target.Coord.Y) * npc.Value.NpcMetadataDistance.Avoid / playerMobDist,
+                npc.Target.Coord.Z + (npc.Coord.Z - npc.Target.Coord.Z) * npc.Value.NpcMetadataDistance.Avoid / playerMobDist
+            );
+
+            npc.Follow(coord);
+            return null;
+        }
+
+        // TODO: Attack
+        // npc.Attack();
+        return null;
+    }
+
+    public override void OnExit(Npc npc) { }
+}
+
+public class BattleStop : AIState
+{
+    public override void OnEnter(Npc npc) { }
+
+    public override AIState? Execute(Npc npc)
+    {
+        PlayerTrigger? playerTrigger = npc.Target!.Value.Triggers.FirstOrDefault(y => y.Key == "battleStop");
+        if (playerTrigger is not null)
+        {
+            playerTrigger.Value = 1;
+            return null;
+        }
+
+        npc.Target.Value.Triggers.Add(new("battleStop", 1));
```

Comment:
This should all be in an AI function, named something like `SetUserValue`

Essentially what I'm getting at with all these is that logic of each AI script shouldn't be in the AI script. This should all be in one/multiple AI files (like how we have Trigger functions).

Having logic per script would increase the tediousness of creating each script. It's like equivalent as if trigger scripts for maps had logic in them. They don't. Instead, they just have conditions, actions, and their respective parameters.